backend/src/openarchiefbeheer/utils/tests/e2e.py

```diff
@@ -55,11 +54,25 @@ def _create_server(self, connections_override=None):
 
 
 class PlaywrightTestCase(StaticLiveServerTestCase):
-    port = settings.E2E_PORT
     fixtures = ["permissions.json"]
     server_thread_class = LiveServerThreadWithReuse
 
+    @classmethod
+    def get_unique_port(cls):
+        base_port = settings.E2E_PORT or 8000
+        try:
+            db_name = connection.settings_dict["NAME"]
+            match_index = re.search(r"\d+", db_name)[0]
+            index = int(match_index)
+            return base_port + index
```

Comment:
Could you maybe comment above it what this is doing?

If I understand correctly, this checks with which database we are connected (the database name has a number, i guess something like `test_openarchiefbeheer_1`, `test_openarchiefbeheer_2`, but the "base" one only `test_openarchiefbeheer`?). If you find a number then the port becomes 8000+1,2... otherwise it errors and returns 8000 (or the E2E port specified).

I wonder: since the frontend needs to know the URL of the backend, how does it know with which live server to talk to?

Reply:
You're correct, I have yet to find a better way to get a unique number per worker that doesn't rely on side effects (database name, PID, etc.) but this seems to be quite a stable approach.

 - Look at a possible number of the test database, or 0 if none was found.
 - Sum it with the default port and compute the final port.
 - Start serving at the final port.

The frontend is aware of this because it's not ware of anything else and falls back to the origin of the frontend itself, which is served by the backend during e2e tests. See the change in `request.ts`.